src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/index.tsx

```diff
@@ -45,29 +51,112 @@ const EditSamplesConfirmationModal = ({
   const [isValid, setIsValid] = useState(false);
   const [metadata, setMetadata] =
     useState<SampleIdToEditMetadataWebform | null>(null);
-
+  const [isContinueButtonActive, setIsContinueButtonActive] =
+    useState<boolean>(false);
+  const [changedMetadata, setChangedMetadata] =
+    useState<SampleIdToEditMetadataWebform | null>(null);
   const { data: namedLocationsData } = useNamedLocations();
   const namedLocations: NamedGisaidLocation[] =
     namedLocationsData?.namedLocations ?? [];
 
-  function structureInitialMetadata(item: Sample): SampleEditMetadataWebform {
-    const i: SampleEditMetadataWebform = pick(
-      item,
-      Object.keys(SAMPLE_EDIT_WEBFORM_METADATA_KEYS_TO_HEADERS)
-    );
-    if (i.collectionLocation) {
-      i.collectionLocation.name = stringifyGisaidLocation(i.collectionLocation);
+  useEffect(() => {
+    // continue button should only be active if the user has metadata
+    // changes and that all form fields are filled out correctly
+    if (changedMetadata && isValid) {
+      setIsContinueButtonActive(true);
+    } else {
+      setIsContinueButtonActive(false);
     }
-    return i;
-  }
+  }, [changedMetadata, isValid]);
 
-  useMemo(() => {
-    const structuredMetadata: SampleIdToEditMetadataWebform = {};
-    checkedSamples.forEach((item) => {
-      structuredMetadata[item.privateId] = structureInitialMetadata(item);
+  const clearState = function () {
+    setChangedMetadata(null);
+    setMetadata(null);
+  };
+
+  const handleClose = function () {
+    clearState();
+    onClose();
+  };
+
+  const updateChangedMetadata = useCallback(
+    (id, sampleMetadata) => {
+      // get all current metadata properties for a sample, if metadata is undefined return empty object
+      const currentMetadata = getMetadataEntryOrEmpty(metadata, id);
+      // combine current metadata with any new metadata requests from user
+      const combinedMetadata: string | boolean | NamedGisaidLocation = {
+        ...currentMetadata,
+        ...sampleMetadata,
+      };
+      const metadataChanges = findMetadataChanges(
+        combinedMetadata,
+        currentMetadata
+      );
+      // save where the metadata has changed
+      setChangedMetadata((prevMetadata) => {
+        const metadataToUpdate = getMetadataEntryOrEmpty(prevMetadata, id);
+        const updatedMetadata = { ...metadataToUpdate, ...metadataChanges };
+        return { ...prevMetadata, [id]: updatedMetadata };
+      });
+    },
+    [metadata]
+  );
+
+  const handleRowMetadata_ = (id: string, sampleMetadata: Metadata) => {
+    updateChangedMetadata(id, sampleMetadata);
+    setMetadata((prevMetadata) => {
+      return { ...prevMetadata, [id]: sampleMetadata };
     });
-    setMetadata(structuredMetadata);
-  }, [checkedSamples]);
+  };
+
+  const handleRowMetadata = useCallback(handleRowMetadata_, [
+    updateChangedMetadata,
+  ]);
+
+  // TODO: update value type to be something other than undefined
```

Comment:
https://app.shortcut.com/genepi/story/193877/resolve-crud-v1-type-todo

Reply:
thanks for making the ticket!